plugins/radar-android-google-sleep/src/main/java/org.radarbase.passive.google.sleep/GoogleSleepManager.kt

```diff
@@ -135,13 +148,15 @@ class GoogleSleepManager(context: GoogleSleepService) : AbstractSourceManager<Go
 
     override fun onClose() {
         sleepHandler.stop {
-           unRegisterFromSleepData()
+            unRegisterSleepReceiver()
+            unRegisterFromSleepData()
```

Comment:
Preferably change to the other way around -- first unregister the sleep client and then the receiver.